tests/unit_tests/ensemble_evaluator/ensemble_evaluator_utils.py

```diff
@@ -194,11 +184,11 @@ def evaluate(self, config):
     def start(self):
         self._eval_thread.start()
 
-    def _shouldFailJob(self, real, step, job):
-        return (real, 0, step, job) in self.fail_jobs
+    def _shouldFailJob(self, real, job):
```

Comment:
This function is used only once, so I think it can be inlined instead.

Reply:
this `self.fail_jobs` list is never updated so the if-sentence using this is probably never hit. This is potentially something unfinished in the test code, and fixing it or removing it should be its own PR.